packages/emitter-framework/src/python/components/protocol-declaration/callable-parameters.tsx

```diff
@@ -0,0 +1,51 @@
+import { TypeExpression } from "#python/components/type-expression/type-expression.js";
+import * as py from "@alloy-js/python";
+import type { Operation } from "@typespec/compiler";
+import { useTsp } from "../../../core/context/tsp-context.js";
+
+export interface CallableParametersProps {
+  /**
+   * The operation to extract parameters from.
+   */
+  type: Operation;
+}
+
+/**
+ * Builds parameter descriptors for a callable (method/function) from an Operation's parameters.
+ *
+ * Iterates over the operation's parameters model and creates ParameterDescriptor objects
+ * with name, type expression, and default value (None for optional parameters).
+ *
+ * @returns Array of ParameterDescriptor objects for use with py.MethodDeclaration or similar.
+ *
+ * @example
+ * ```tsx
+ * const params = CallableParameters({ type: operation });
+ * <py.MethodDeclaration name="foo" parameters={params} />
+ * ```
+ */
+export function CallableParameters(props: CallableParametersProps): py.ParameterDescriptor[] {
+  const { $ } = useTsp();
+  const paramsModel = props.type.parameters;
+
+  if (!paramsModel) {
+    return [];
+  }
+
+  const parameters: py.ParameterDescriptor[] = [];
+
+  try {
+    for (const prop of $.model.getProperties(paramsModel).values()) {
+      parameters.push({
+        name: prop.name,
+        type: <TypeExpression type={prop.type} />,
+        // Optional parameters without explicit defaults get `= None`
+        ...(prop.optional ? { default: <>None</> } : {}),
+      });
+    }
+  } catch {
+    // If getProperties fails, return empty array
```

Comment:
When does this happen? Should avoid this error being possible rather than handling it unless this is working around some bug?